frontend/packages/data-portal/app/graphql/getDatasetByIdDiffer.ts

```diff
@@ -0,0 +1,278 @@
+import { diff } from 'deep-object-diff'
+
+import { GetDatasetByIdQuery } from 'app/__generated__/graphql'
+import {
+  Annotation_File_Shape_Type_Enum,
+  GetDatasetByIdV2Query,
+  Sample_Type_Enum,
+  Tiltseries_Microscope_Manufacturer_Enum,
+} from 'app/__generated_v2__/graphql'
+
+/* eslint-disable no-console, no-param-reassign */
+export function logIfHasDiff(
+  url: string,
+  v1: GetDatasetByIdQuery,
+  v2: GetDatasetByIdV2Query,
+): void {
+  console.log('Checking for dataset query diffs')
+
+  v2 = structuredClone(v2)
+
+  // Counts not used.
+  for (const run of v2.runs) {
+    for (const annotationsAggregate of run.annotationsAggregate?.aggregate ??
+      []) {
+      delete annotationsAggregate.count
+    }
+    // Consistent sort order.
+    run.annotationsAggregate?.aggregate?.sort((groupA, groupB) =>
+      groupA.groupBy!.objectName!.localeCompare(groupB.groupBy!.objectName!),
+    )
+  }
+  for (const annotationsAggreate of v2.annotationsAggregate.aggregate ?? []) {
+    delete annotationsAggreate.count
+  }
+  for (const annotationShapesAggregate of v2.annotationShapesAggregate
+    .aggregate ?? []) {
+    delete annotationShapesAggregate.count
+  }
+  for (const tiltseriesAggregate of v2.tiltseriesAggregate.aggregate ?? []) {
+    delete tiltseriesAggregate.count
+  }
+  // Consistent sort order.
+  v2.annotationsAggregate.aggregate?.sort((groupA, groupB) =>
+    groupA.groupBy!.objectName!.localeCompare(groupB.groupBy!.objectName!),
+  )
+  v2.annotationShapesAggregate.aggregate?.sort((groupA, groupB) =>
+    groupA.groupBy!.shapeType!.localeCompare(groupB.groupBy!.shapeType!),
+  )
+  v2.tiltseriesAggregate.aggregate?.sort(
+    (groupA, groupB) =>
+      groupA.groupBy!.tiltSeriesQuality! - groupB.groupBy!.tiltSeriesQuality!,
+  )
+
+  const v1Transformed: GetDatasetByIdV2Query = {
```

Comment:
Good question. I guess the relevant function there is `remapAPI()`. I don't think it's that useful in this case because that function is just converting keys which is more explicit to just do here directly. Also defining the object here in one statement lets us make sure it's typed to `GetDatasetByIdV2Query` which I think the way `remapAPI()` is implemented right now can't guarantee.